web-app/js/portal/ui/openlayers/control/SpatialConstraint.js

```diff
@@ -155,6 +156,10 @@ Portal.ui.openlayers.control.SpatialConstraint = Ext.extend(OpenLayers.Control.D
         }
     },
 
+    getCleanGeometry: function(geometry) {
+        return new OpenLayers.Geometry.fromWKT(geometry.toWkt());
```

Comment:
So is this translating the geometry to WKT, normalizing it, and then returning it?

Ugly, but this is what we can do I guess.